docs/core/docker/docker-basics-dotnet-core.md

```diff
@@ -0,0 +1,287 @@
+
+You can build both Windows and Linux containers with [multi-arch based tags](https://github.com/dotnet/announcements/issues/14).
+
+## Your first .NET Core Docker app
+
+### Prerequisites
+
+To complete this tutorial:
+
+#### .NET Core 2.0 SDK
+
+* Install [.NET Core SDK 2.0](https://www.microsoft.com/net/core).
+
+* Install your favorite code editor, if you haven't already.
+
+> [!TIP]
+> Need to install a code editor? Try [Visual Studio](https://visualstudio.com/downloads)!
+
+#### Installing Docker Client
+
+Install [Docker 17.06](https://docs.docker.com/release-notes/docker-ce/) or later of the Docker client.
+
+The Docker client can be installed in:
+
+* Linux distributions
+
+   * [CentOS](https://www.docker.com/docker-centos-distribution)
+
+   * [Debian](https://www.docker.com/docker-debian)
+
+   * [Fedora](https://www.docker.com/docker-fedora)
+
+   * [Ubuntu](https://www.docker.com/docker-ubuntu)
+
+* [macOS](https://docs.docker.com/docker-for-mac/)
+
+* [Windows](https://docs.docker.com/docker-for-windows/).
+
+### Create a .NET Core 2.0 console app for Dockerization
+
+Open a command prompt and create a folder named *Hello*. Navigate to the folder you created and type the following commands:
+
+```console
+dotnet new console
+dotnet run
+```
+
+Let's do a quick walkthrough:
+
+1. `$ dotnet new console`
+
+   [`dotnet new`](../tools/dotnet-new.md) creates an up-to-date `Hello.csproj` project file with the dependencies necessary to build a console app.  It also creates a `Program.cs`, a basic file containing the entry point for the application.
+   
+   `Hello.csproj`:
+
+   The project file specifies everything that's needed to restore dependencies and build the program.
+
+   * The `OutputType` tag specifies that we're building an executable, in other words a console application.
+   * The `TargetFramework` tag specifies what .NET implementation we're targeting. In an advanced scenario, you can specify multiple target frameworks and build to the specified frameworks in a single operation. In this tutorial, we build for .NET Core 2.0.
+
+   `Program.cs`:
+
+   The program starts by `using System`. This statement means, "Bring everything in the `System` namespace into scope for this file." The `System` namespace includes basic constructs such as `string`, or numeric types.
+
+   We then define a namespace called `Hello`. You can change namespace to anything you want. A class named `Program` is defined within that namespace, with a `Main` method that takes an array of strings as its argument. This array contains the list of arguments passed in when the compiled program is called. In our example, the program only writes "Hello World!" to the console.
+
+2. `$ dotnet restore`
+
+   In .NET Core 2.x, **dotnet new** runs the [`dotnet restore`](../tools/dotnet-restore.md) command. **Dotnet restore** restores the tree of dependencies with a [NuGet](https://www.nuget.org/)(.NET package manager) call. 
```

Comment:
`dotnet restore` is triggered by more than just `new`.  It is now a command that is implicitly invoked with any command that requires it.

Reply:
Yes, that's true. I was speaking in the context of the .NET Core 2.0 changes to dotnet new. I will include a note about the changes around dotnet restore as well.

Reply:
Actually, the linked article about dotnet restore talks about the implicit invocation quite well. Upon reflection, I think that will be a better choice. Ok with that?